### PR TITLE
[FD]Formatting Frontier Defense Gamemode Files

### PR DESCRIPTION
**Pull request by RoyalBlue1**
Formats script files of frontier defense to be more in line with other script files

### CODE REVIEW

File: Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_fd.nut [hunk 1]
```diff
@@ -52,27 +52,27 @@ void function GamemodeFD_Init()
...
 
 	//general Callbacks
-	AddCallback_EntitiesDidLoad(LoadEntities)
-	AddCallback_GameStateEnter(eGameState.Prematch,FD_createHarvester)
-	AddCallback_GameStateEnter( eGameState.Playing,startMainGameLoop)
-	AddCallback_OnRoundEndCleanup(FD_NPCCleanup)
-	AddCallback_OnClientConnected(GamemodeFD_InitPlayer)
+	AddCallback_EntitiesDidLoad( LoadEntities )
+	AddCallback_GameStateEnter( eGameState.Prematch,FD_createHarvester )
+	AddCallback_GameStateEnter( eGameState.Playing,startMainGameLoop )
```
Comment (ASpoonPlaysGames) [inline Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_fd.nut:70]:
```suggestion
	AddCallback_GameStateEnter( eGameState.Playing, startMainGameLoop )
```

File: Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_fd.nut [hunk 2]
```diff
@@ -83,12 +83,12 @@ void function GamemodeFD_Init()
...
-	AddCallback_OnNPCKilled(OnNpcDeath)
-	AddCallback_OnPlayerKilled(GamemodeFD_OnPlayerKilled)
+	AddCallback_OnNPCKilled( OnNpcDeath )
+	AddCallback_OnPlayerKilled( GamemodeFD_OnPlayerKilled )
 	
 	//Command Callbacks
-	AddClientCommandCallback("FD_ToggleReady",ClientCommandCallbackToggleReady)
-	AddClientCommandCallback("FD_UseHarvesterShieldBoost",useShieldBoost)
+	AddClientCommandCallback( "FD_ToggleReady",ClientCommandCallbackToggleReady )
+	AddClientCommandCallback( "FD_UseHarvesterShieldBoost",useShieldBoost )
```
Comment (ASpoonPlaysGames) [inline Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_fd.nut:91]:
```suggestion
	AddClientCommandCallback( "FD_ToggleReady", ClientCommandCallbackToggleReady )
	AddClientCommandCallback( "FD_UseHarvesterShieldBoost", useShieldBoost )
```

File: Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_fd.nut [hunk 3]
```diff
@@ -123,87 +123,87 @@ void function FD_TeamReserveDepositOrWithdrawCallback(entity player, string acti
...
 			continue
 
-		if (deaths == 1) // only one pilot died
+		if ( deaths == 1 ) // only one pilot died
 			PlayFactionDialogueToPlayer( "fd_singlePilotDown", player )
-		else if (deaths > 1 && deaths < militiaplayers.len() - 1) // multiple pilots died but at least one alive
+		else if ( deaths > 1 && deaths < militiaplayers.len() - 1 ) // multiple pilots died but at least one alive
 			PlayFactionDialogueToPlayer( "fd_multiPilotDown", player )
-		else if (deaths == militiaplayers.len() - 1) // ur shit out of luck ur the only survivor
+		else if (d eaths == militiaplayers.len() - 1 ) // ur shit out of luck ur the only survivor
```
Comment (ASpoonPlaysGames) [inline Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_fd.nut:145]:
```suggestion
		else if ( deaths == militiaplayers.len() - 1 ) // ur shit out of luck ur the only survivor
```

File: Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_fd.nut [hunk 4]
```diff
@@ -259,15 +259,18 @@ void function OnNpcDeath( entity victim, entity attacker, var damageInfo )
...
+void function RateSpawnpoints_FD( int _0, array<entity> _1, int _2, entity _3 )
+{
+
+}
 
-bool function useShieldBoost(entity player,array<string> args)
+bool function useShieldBoost( entity player, array<string> args )
 {
-	if((GetGlobalNetTime("FD_harvesterInvulTime")<Time())&&(player.GetPlayerNetInt("numHarvesterShieldBoost")>0))
+	if( ( GetGlobalNetTime( "FD_harvesterInvulTime" ) < Time() ) && ( player.GetPlayerNetInt("numHarvesterShieldBoost") > 0 ) )
```
Comment (ASpoonPlaysGames) [inline Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_fd.nut:269]:
```suggestion
	if( ( GetGlobalNetTime( "FD_harvesterInvulTime" ) < Time() ) && ( player.GetPlayerNetInt( "numHarvesterShieldBoost" ) > 0 ) )
```
we should be consistent on the same line, either remove spaces for the first one or add spaces to the second

File: Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_fd.nut [hunk 5]
```diff
@@ -259,15 +259,18 @@ void function OnNpcDeath( entity victim, entity attacker, var damageInfo )
...
 	{
 		fd_harvester.harvester.SetShieldHealth(fd_harvester.harvester.GetShieldHealthMax())
```
Comment (ASpoonPlaysGames) [inline Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_fd.nut:271]:
```suggestion
		fd_harvester.harvester.SetShieldHealth( fd_harvester.harvester.GetShieldHealthMax() )
```

File: Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_fd.nut [hunk 6]
```diff
@@ -284,22 +287,22 @@ void function mainGameLoop()
 	startHarvester()
 
 	bool showShop = false
-	for(int i = GetGlobalNetInt("FD_currentWave");i<waveEvents.len();i++)
+	for(int i = GetGlobalNetInt( "FD_currentWave" ); i < waveEvents.len(); i++)
```
Comment (ASpoonPlaysGames) [inline Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_fd.nut:290]:
```suggestion
	for( int i = GetGlobalNetInt( "FD_currentWave" ); i < waveEvents.len(); i++ )
```

File: Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_fd.nut [hunk 7]
```diff
@@ -311,211 +314,212 @@ void function mainGameLoop()
...
+			case( eFD_AITypeIDs.SCORCH ):
+			case( eFD_AITypeIDs.TONE ):
+			case( eFD_AITypeIDs.ION ):
+			case( eFD_AITypeIDs.MONARCH ):
+			case( eFD_AITypeIDs.LEGION ):
+			case( eFD_AITypeIDs.TITAN_SNIPER ):
 				npcs[eFD_AITypeIDs.TITAN]+=e.spawnEvent.spawnAmount
 				break
 			default:
 				npcs[e.spawnEvent.spawnType]+=e.spawnEvent.spawnAmount
```
Comment (ASpoonPlaysGames) [inline Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_fd.nut:360]:
```suggestion
				npcs[eFD_AITypeIDs.TITAN] += e.spawnEvent.spawnAmount
				break
			default:
				npcs[e.spawnEvent.spawnType] += e.spawnEvent.spawnAmount
```

File: Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_fd.nut [hunk 8]
```diff
@@ -311,211 +314,212 @@ void function mainGameLoop()
...
 		{
-			if(arrayValue==-1)
+			if( arrayValue == -1 )
 			{
 				ret[index] = key
 				keyIsSet = true
 				break
 			}
-			if(npcs[ret[lowestArrayIndex]]>npcs[ret[index]])
+			if( npcs[ret[lowestArrayIndex]] > npcs[ret[index]] )
```
Comment (ASpoonPlaysGames) [inline Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_fd.nut:378]:
```suggestion
			if( npcs[ ret[ lowestArrayIndex ] ] > npcs[ ret[ index ] ] )
```
I'd say for nested [] we should use spaces for better readability

File: Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_fd.nut [hunk 9]
```diff
@@ -524,73 +528,73 @@ bool function runWave(int waveIndex,bool shouldDoBuyTime)
...
 				highestDamage[2] = damage
 				highestDamageSource[2] = index
 			}
 		}
 
-		foreach(entity player in GetPlayerArray())
+		foreach( entity player in GetPlayerArray() )
 		{
-			Remote_CallFunction_NonReplay(player,"ServerCallback_FD_DisplayHarvesterKiller",GetGlobalNetInt("FD_restartsRemaining"),getHintForTypeId(highestDamageSource[0]),highestDamageSource[0],highestDamage[0]/totalDamage,highestDamageSource[1],highestDamage[1]/totalDamage,highestDamageSource[2],highestDamage[2]/totalDamage)
+			Remote_CallFunction_NonReplay( player, "ServerCallback_FD_DisplayHarvesterKiller", GetGlobalNetInt("FD_restartsRemaining"), getHintForTypeId(highestDamageSource[0]), highestDamageSource[0], highestDamage[0] / totalDamage, highestDamageSource[1], highestDamage[1] / totalDamage , highestDamageSource[2], highestDamage[2] / totalDamage )
```
Comment (ASpoonPlaysGames) [inline Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_fd.nut:547]:
```suggestion
			Remote_CallFunction_NonReplay( player, "ServerCallback_FD_DisplayHarvesterKiller", GetGlobalNetInt( "FD_restartsRemaining" ), getHintForTypeId( highestDamageSource[0] ), highestDamageSource[0], highestDamage[0] / totalDamage, highestDamageSource[1], highestDamage[1] / totalDamage , highestDamageSource[2], highestDamage[2] / totalDamage )
```

File: Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_fd.nut [hunk 10]
```diff
@@ -626,33 +630,33 @@ bool function runWave(int waveIndex,bool shouldDoBuyTime)
...
-			AddPlayerScore(player,"FDDidntDie")
+			AddPlayerScore( player, "FDDidntDie" )
 			player.AddToPlayerGameStat( PGS_ASSAULT_SCORE, FD_SCORE_DIDNT_DIE )
 		}
-		AddMoneyToPlayer(player,100)
-		EmitSoundOnEntityOnlyToPlayer(player,player,"HUD_MP_BountyHunt_BankBonusPts_Deposit_Start_1P")
-		if(highestScore<file.players[player].scoreThisRound)
+		AddMoneyToPlayer( player, 100 )
+		EmitSoundOnEntityOnlyToPlayer( player, player, "HUD_MP_BountyHunt_BankBonusPts_Deposit_Start_1P" )
+		if( highestScore<file.players[player].scoreThisRound )
```
Comment (ASpoonPlaysGames) [inline Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_fd.nut:659]:
```suggestion
		if( highestScore < file.players[player].scoreThisRound )
```

File: Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_fd.nut [hunk 11]
```diff
@@ -661,48 +665,50 @@ bool function runWave(int waveIndex,bool shouldDoBuyTime)
...
+	foreach( entity player in GetPlayerArray() )
 	{
 
-		if(!file.havesterWasDamaged)
+		if( !file.havesterWasDamaged )
 		{
-			AddPlayerScore(player,"FDTeamFlawlessWave")
-			AddMoneyToPlayer(player,100)
+			AddPlayerScore( player, "FDTeamFlawlessWave" )
+			AddMoneyToPlayer( player, 100)
```
Comment (ASpoonPlaysGames) [inline Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_fd.nut:683]:
```suggestion
			AddMoneyToPlayer( player, 100 )
```

File: Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_fd.nut [hunk 12]
```diff
@@ -748,21 +754,21 @@ void function OnHarvesterDamaged(entity harvester, var damageInfo)
...
-		if (healthpercent <= 50 && oldhealthpercent > 50)
+		if ( healthpercent <= 50 && oldhealthpercent > 50 )
 			PlayFactionDialogueToTeam( "fd_baseHealth50", TEAM_MILITIA )
 
-		if (healthpercent <= 25 && oldhealthpercent > 25)
+		if ( healthpercent <= 25 && oldhealthpercent > 25 )
 			PlayFactionDialogueToTeam( "fd_baseHealth25", TEAM_MILITIA )
 
-		if (healthpercent <= 10)
+		if  (healthpercent <= 10 )
```
Comment (ASpoonPlaysGames) [inline Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_fd.nut:766]:
```suggestion
		if ( healthpercent <= 10 )
```

File: Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_fd.nut [hunk 13]
```diff
@@ -848,22 +854,22 @@ void function HarvesterThink()
 
 			if ( newShieldHealth >= harvester.GetShieldHealthMax() )
 			{
-				StopSoundOnEntity(harvester,"coop_generator_shieldrecharge_resume")
-				harvester.SetShieldHealth(harvester.GetShieldHealthMax())
-				EmitSoundOnEntity(harvester,"coop_generator_shieldrecharge_end")
-				if (GetGlobalNetBool("FD_waveActive"))
+				StopSoundOnEntity( harvester, "coop_generator_shieldrecharge_resume" )
+				harvester.SetShieldHealth( harvester.GetShieldHealthMax())
```
Comment (ASpoonPlaysGames) [inline Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_fd.nut:858]:
```suggestion
				harvester.SetShieldHealth( harvester.GetShieldHealthMax() )
```

File: Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_fd.nut [hunk 14]
```diff
@@ -1000,54 +1008,56 @@ void function FD_createHarvester()
...
 					break
 				case"info_fd_ai_position":
-					AddStationaryAIPosition(info_target.GetOrigin(),int(info_target.kv.aiType))
+					AddStationaryAIPosition( info_target.GetOrigin(), int( info_target.kv.aiType ) )
 					break
 				case"info_fd_route_node":
-					routeNodes.append(info_target)
+					routeNodes.append( info_target )
 					break
 				case"info_fd_smoke_screen":
```
Comment (ASpoonPlaysGames) [inline Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_fd.nut:1044]:
```suggestion
				case "info_fd_smoke_screen":
```

File: Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_fd.nut [hunk 15]
```diff
@@ -1000,54 +1008,56 @@ void function FD_createHarvester()
...
 					file.harvester_info = info_target
 					break
 				case"info_fd_mode_model":
 					entity prop = CreatePropDynamic( info_target.GetModelName(), info_target.GetOrigin(), info_target.GetAngles(), 6 )
 					break
 				case"info_fd_ai_position":
-					AddStationaryAIPosition(info_target.GetOrigin(),int(info_target.kv.aiType))
+					AddStationaryAIPosition( info_target.GetOrigin(), int( info_target.kv.aiType ) )
 					break
 				case"info_fd_route_node":
```
Comment (ASpoonPlaysGames) [inline Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_fd.nut:1041]:
```suggestion
				case "info_fd_route_node":
```

File: Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_fd.nut [hunk 16]
```diff
@@ -1000,54 +1008,56 @@ void function FD_createHarvester()
...
+		{
+			switch( info_target.kv.editorclass )
+			{
 				case"info_fd_harvester":
 					file.harvester_info = info_target
 					break
 				case"info_fd_mode_model":
 					entity prop = CreatePropDynamic( info_target.GetModelName(), info_target.GetOrigin(), info_target.GetAngles(), 6 )
 					break
 				case"info_fd_ai_position":
```
Comment (ASpoonPlaysGames) [inline Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_fd.nut:1038]:
```suggestion
				case "info_fd_ai_position":
```

File: Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_fd.nut [hunk 17]
```diff
@@ -1000,54 +1008,56 @@ void function FD_createHarvester()
...
-		if(info_target.HasKey("editorclass")){
-			switch(info_target.kv.editorclass){
+		if( info_target.HasKey( "editorclass" ) )
+		{
+			switch( info_target.kv.editorclass )
+			{
 				case"info_fd_harvester":
 					file.harvester_info = info_target
 					break
 				case"info_fd_mode_model":
```
Comment (ASpoonPlaysGames) [inline Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_fd.nut:1035]:
```suggestion
				case "info_fd_mode_model":
```

File: Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_fd.nut [hunk 18]
```diff
@@ -1000,54 +1008,56 @@ void function FD_createHarvester()
...
 		if ( GameModeRemove( info_target ) )
 			continue
 
-		if(info_target.HasKey("editorclass")){
-			switch(info_target.kv.editorclass){
+		if( info_target.HasKey( "editorclass" ) )
+		{
+			switch( info_target.kv.editorclass )
+			{
 				case"info_fd_harvester":
```
Comment (ASpoonPlaysGames) [inline Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_fd.nut:1032]:
```suggestion
				case "info_fd_harvester":
```

File: Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_fd.nut (hunk 8 above)
Comment (RoyalBlue1) [reply Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_fd.nut:378]: i think the inner ones should stay without

File: Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_fd.nut (hunk 12 above)
Comment (RoyalBlue1) [reply Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_fd.nut:766]:
```suggestion
		if( healthpercent <= 10 )
```

Comment (ASpoonPlaysGames) [reply Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_fd.nut]: I don't really think it matters, Respawn aren't consistent with this :/
image
image

File: Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_fd.nut (hunk 8 above)
Comment (ASpoonPlaysGames) [reply Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_fd.nut:378]: yeah maybe
```suggestion
			if( npcs[ ret[lowestArrayIndex] ] > npcs[ ret[index] ] )
```

File: Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_fd.nut [hunk 19]
```diff
@@ -83,12 +83,12 @@ void function GamemodeFD_Init()
...
+	AddCallback_OnPlayerKilled( GamemodeFD_OnPlayerKilled )
 	
 	//Command Callbacks
-	AddClientCommandCallback("FD_ToggleReady",ClientCommandCallbackToggleReady)
-	AddClientCommandCallback("FD_UseHarvesterShieldBoost",useShieldBoost)
+	AddClientCommandCallback( "FD_ToggleReady", ClientCommandCallbackToggleReady )
+	AddClientCommandCallback( "FD_UseHarvesterShieldBoost", useShieldBoost )
 
 	//shop Callback
 	SetBoostPurchaseCallback(FD_BoostPurchaseCallback)
```
Comment (x3Karma) [inline Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_fd.nut:94]:
```suggestion
	SetBoostPurchaseCallback( FD_BoostPurchaseCallback )
```

Comment (x3Karma) [reply Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_fd.nut:94]: Line 95 also needs spacing but github doesn't allowing reviewing on code that isn't near the modified diff.

**Pull request merged**